projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java

```diff
@@ -349,20 +397,12 @@ public Response authorizeNewUser(@HeaderParam("_protocol") String protocol,
 					case KEY_ALREADY_EXIST:
 						return exceptionToResponse(new DuplicateAPIKeyException(
 								username), applicationID, "not-resolved", username);
-					case KEY_DOES_NOT_EXIST:
-						return exceptionToResponse(new APIKeyDoesNotExistException(
-								username), applicationID, "not-resolved", username);
 					default:
-						JsonObject retVal = new JsonObject();
-						retVal.add("error",new JsonPrimitive(apiKeyManagerException.getMessage()));
-						return Response.status(500).entity(retVal.toString()).
-								type("application/json").build();
+						return exceptionToResponse(apiKeyManagerException.getMessage());
 				}
 			} catch (Exception e) {
-				JsonObject retVal = new JsonObject();
-				retVal.add("error",new JsonPrimitive(e.getMessage()));
-				return Response.status(500).entity(retVal.toString()).
-						type("application/json").build();
+				return exceptionToResponse(e.getMessage());
```

Comment:
Shouldn't we log something here before returning, with log.error() or do you think that is too verbose?